_posts/2023-07-09-infra-migration.md

```diff
@@ -0,0 +1,332 @@
+---
+title: The 2023 infrastructure migration
+author: TrueBrain
+---
+
+This post is part of a two-part series, of which this is the second.
+In the [first post](/news/2023/07/08/our-infrastructure), we went into a bit into the infrastructure that runs OpenTTD, from BaNaNaS to our main website.
+In this post, we will explain a bit about the migration we just did to get to this infrastructure.
+
+After over 2 months of work, I am happy to announce we finished ([another](https://www.openttd.org/news/2020/12/01/infrastructure-migration-finished)) infrastructure migration.
+Today is the day I removed the last few DNS entries pointing to [AWS](https://aws.amazon.com/)'s DNS servers, and I am proud to mention that (almost) all traffic is now routed via [Cloudflare](https://cloudflare.com/) (and we aren't even receiving sponsoring to say so).
+
+In this post I want to take you with me why this migration was needed, what the benefits are, and why you possibly care.
+But in short summary:
+
+- (Much) smaller monthly bill as AWS charges insane amounts for bandwidth.
+- Faster download speeds for you (ranging from the in-game content service to downloading the game from our website).
+- Easier maintainability of our infrastructure with thanks to [Pulumi](https://pulumi.com/).
+
+This will be a bit nerdy, so if you like these kind of things, continue the read!
+
+<!-- more -->
+
+## History
+
+As with every migration post, it starts with a bit of history.
+Just to catch you up to a few months ago.
+
+In 2020 we started the migration from our own dedicated hosts to AWS.
+Back then, AWS was offering Open Source credits, which meant we didn't have to worry so much about money.
+Additionally, they had everything we needed, and it seemed like a good fit.
+
+This has been a very good call; not only did I personally have far less worries about things like security, stability, etc, we also had the fewest interruption till date.
+More than 99.99% uptime over all our services since then.
+This means it has been very unlikely you couldn't download the game, or use the in-game content service, etc.
+Even big companies would be very happy with 99.99%, but given we run this with only a handful of people in their free time ... most excellent, if you ask me.
+
+The fun thing is, with such uptime, when we got a report of a problem: it is very likely to be a user-problem, and not a backend-problem.
+This is a great spot to be in!
+
+## Problems on the horizon
+
+But, for more than a year now, there have been some issues that were hard to deal with.
+Let's explain a few in a bit more detail.
+
+### AWS's bandwidth price
+
+The Open Source credits ran out after the first year; and where we hoped (and somewhat assumed) the AWS bandwidth prices would go down over time, they did not.
+For those who don't know, AWS charges 9 dollar cents per GB of bandwidth.
```

Comment:
Is this a common way to specify US Dollar cents? I've never seen it (as an American where all cents are assumed to be USD).

All your other dollar amounts use the world dollar, so I don't know if you'd want to change it, but I generally write money with the $ sign, so $0.09.

Dunno what to do here, feel free to leave it as-is. 🤷 😄

Reply:
changed them all in USD.